### PR TITLE
Reconfigure tox to reuse same envs between unit and functional

### PR DESCRIPTION
**Pull request by ssbarnea**
Removes 9 environments by reusing them across unit and functional
testing commands as they have identical dependencies.

Fixes: #1546

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -1,15 +1,16 @@
...
+    py{27,36,37}-ansible{24,25,26}-functional
+    py{27,36,27}-ansible{24,25,26}-unit
     py{27,36,37}-lint
     format-check
     doc
 skipdist = True
 skip_missing_interpreters=True
+# sitepackages improve speed and footprint at small risk of missing to spot
+# undeclared dependencies which are already installed on the system.
+sitepackages=True
```
Comment (webknjaz) [inline tox.ini]: I don't think we should do this. Tests are already unstable. No more cumulative risks, please.

File: tox.ini [hunk 2]
```diff
@@ -1,15 +1,16 @@
-
 [tox]
 minversion = 1.8
 envlist =
-    py{27}-ansible{24,25,26}-functional
-    py{27,36,37}-ansible{24,25,26}-unit
-    py{36,37}-ansible{24,25,26}-functional
+    py{27,36,37}-ansible{24,25,26}-functional
```
Comment (webknjaz) [inline tox.ini]: Please wait until I extract this from my PR. I've got it extended with a few more.

**Comment by webknjaz** [comment]
@ssbarnea please rebase this after #1557 merged

**Comment by webknjaz** [comment]
@ssbarnea you can probably start rebasing this

File: tox.ini [hunk 3]
```diff
@@ -1,15 +1,16 @@
...
+    py{27,36,27}-ansible{24,25,26}-unit
```
Comment (webknjaz) [inline tox.ini]: you can also collapse the last factor as `{functional,unit}`

File: tox.ini (hunk 2 above)
Comment (webknjaz) [inline tox.ini]:
```suggestion
    py{27,36,37}-ansible{24,25,26}-{functional,unit}
```

File: tox.ini (hunk 3 above)
Comment (webknjaz) [inline tox.ini]:
```suggestion
```

**Comment by themr0c** [comment]
@ssbarnea Can you please rebase, so we can have a view if the proposal is still compatible with current tox.ini?

File: tox.ini (hunk 1 above)
Comment (ssbarnea) [reply tox.ini]: sure, better to do this isolated if we do it.

**Pull request closed without merging**